applications/teletype/src/templates/teletype_low_balance.erl

```diff
@@ -0,0 +1,179 @@
+
+-define(TEMPLATE_ID, <<"low_balance">>).
+-define(MOD_CONFIG_CAT, <<(?NOTIFY_CONFIG_CAT)/binary, ".", (?TEMPLATE_ID)/binary>>).
+
+-define(TEMPLATE_MACROS
+        ,wh_json:from_list(
+           [?MACRO_VALUE(<<"user.first_name">>, <<"first_name">>, <<"First Name">>, <<"First Name">>)
+            ,?MACRO_VALUE(<<"user.last_name">>, <<"last_name">>, <<"Last Name">>, <<"Last Name">>)
+            | ?SERVICE_MACROS ++ ?ACCOUNT_MACROS
+           ])
+       ).
+
+-define(TEMPLATE_TEXT, <<"The account \"{{account.name}}\" has less than {{threshold}} of credit remaining.\nIf the account runs out of credit it will not be able to make or receive per-minute calls.\nThe current balance is: {{current_balance}}\n\nAccount ID: {{account.id}}">>).
+-define(TEMPLATE_HTML, <<"<html><body><h2>The account \"{{account.name}}\" has less than {{threshold}} of credit remaining.</h2><p>Current Balance: {{current_balance}}</p><p>If the account runs out of credit it will not be able to make or receive per-minute calls.</body></html>">>).
+-define(TEMPLATE_SUBJECT, <<"Account {{account.name}} is running out of credit">>).
+-define(TEMPLATE_CATEGORY, <<"account">>).
+-define(TEMPLATE_NAME, <<"Low Balance">>).
+
+-define(TEMPLATE_TO, ?CONFIGURED_EMAILS(?EMAIL_ORIGINAL)).
+-define(TEMPLATE_FROM, teletype_util:default_from_address(?MOD_CONFIG_CAT)).
+-define(TEMPLATE_CC, ?CONFIGURED_EMAILS(?EMAIL_SPECIFIED, [])).
+-define(TEMPLATE_BCC, ?CONFIGURED_EMAILS(?EMAIL_SPECIFIED, [])).
+-define(TEMPLATE_REPLY_TO, teletype_util:default_reply_to(?MOD_CONFIG_CAT)).
+
+-spec init() -> 'ok'.
+init() ->
+    wh_util:put_callid(?MODULE),
+
+    teletype_util:init_template(?TEMPLATE_ID, [{'macros', ?TEMPLATE_MACROS}
+                                               ,{'text', ?TEMPLATE_TEXT}
+                                               ,{'html', ?TEMPLATE_HTML}
+                                               ,{'subject', ?TEMPLATE_SUBJECT}
+                                               ,{'category', ?TEMPLATE_CATEGORY}
+                                               ,{'friendly_name', ?TEMPLATE_NAME}
+                                               ,{'to', ?TEMPLATE_TO}
+                                               ,{'from', ?TEMPLATE_FROM}
+                                               ,{'cc', ?TEMPLATE_CC}
+                                               ,{'bcc', ?TEMPLATE_BCC}
+                                               ,{'reply_to', ?TEMPLATE_REPLY_TO}
+                                              ]).
+
+-spec handle_low_balance(wh_json:object(), wh_proplist()) -> 'ok'.
+handle_low_balance(JObj, _Props) ->
+    'true' = wapi_notifications:low_balance_v(JObj),
+    wh_util:put_callid(JObj),
+
+    %% Gather data for template
+    DataJObj0 = wh_json:normalize(wh_api:remove_defaults(JObj)),
+
+    ServerID = wh_json:get_first_defined([<<"server_id">>, <<"Server-ID">>], JObj),
+    DataJObj = wh_json:set_value(<<"Server-ID">>, ServerID, DataJObj0),
+
```

Comment:
why not use just wh_json:normalize ?

server_id will be handled in send_update if that's the case.